### PR TITLE
[Perf]: Improve query plan for "delete from TABLE_NAME" without any conditions

### PR DESCRIPTION
**Pull request by noorall**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [x] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #12515

## What this PR does / why we need it:
When executing the DELETE without condition, the join operation on auxiliary tables is removed, and table locks are used to reduce the cost of locking.

### CODE REVIEW

**Review by ouyuanning — COMMENTED** [review]
better to add some BVT case.
eg:
delete normal table with/without where
delete table with unique key, 2nd key, foreign key with/without where
delete multi tables with/without where

File: pkg/sql/plan/build_delete.go [hunk 1]
```diff
@@ -55,7 +55,8 @@ func buildDelete(stmt *tree.Delete, ctx CompilerContext, isPrepareStmt bool) (*P
 	beginIdx := 0
 	// needLockTable := !tblInfo.isMulti && stmt.Where == nil && stmt.Limit == nil
 	// todo will do not lock table now.
-	needLockTable := false
+	isDeleteWithoutFilters := stmt.Where == nil && len(tblInfo.tableDefs) == 1
```
Comment (ouyuanning) [inline pkg/sql/plan/build_delete.go:58]: should like:
isDeleteWithoutFilters := !tblInfo.isMulti && stmt.Where == nil && stmt.Limit == nil

**Comment by noorall** [comment]
> better to add some BVT case. eg: delete normal table with/without where delete table with unique key, 2nd key, foreign key with/without where delete multi tables with/without where

added

**Review by ouyuanning — CHANGES_REQUESTED** [review]
have some bug for delete big data. need fixed

File: test/distributed/cases/dml/delete/delete.test [hunk 2]
```diff
@@ -216,4 +216,47 @@ insert into t2 values (1,2,1,2);
...
+drop table if exists t4;
+create table t4 (a int, b int, c int);
+insert into t4 (a, b, c) values (1, 100, 1000);
+insert into t4 (a, b, c) values (2, 200, 2000);
+insert into t4 (a, b, c) values (3, 300, 3000);
+delete from t4 where a > 2;
+select * from t4;
+delete from t4;
+select * from t4;
+
```
Comment (ouyuanning) [inline test/distributed/cases/dml/delete/delete.test:256]: 增加几个用例吧
1、有外键的情况
2、有limit 子句的
3、200W数据的表
```
create table t1(a int primary key, b int unique key, c varchar(20) unique key);
insert into t1 select result, result, "a"||result from generate_series(1,2000000) g;
```

**Comment by noorall** [comment]
@aressu1985 @heni02 Please review the test code.

**Review by ouyuanning — APPROVED** [review]

**Review by aressu1985 — APPROVED** [review]

**Pull request merged**